app/models/spree/variant.rb

```diff
@@ -300,7 +300,7 @@ def destruction
     end
 
     def ensure_unit_value
-      return unless product&.variant_unit == "items" && unit_value.nil?
+      return unless (product&.variant_unit == "items" && unit_value.nil?) || unit_value&.nan?
```

Comment:
Great. We should probably add a unit test for this.

I was wondering about another approach as well: If we just add a validation then we still prevent this from happening but we might also notice (as user or in Bugsnag) where this is happening and find out the reason. The finding might be insightful. It might also be boring and we still want this code to automatically set the value. :shrug: 

A validation could be helpful anyway: https://guides.rubyonrails.org/active_record_validations.html#numericality

Reply:
I just tried out adding a validation, like so:

`validates :unit_value, numericality: true`

and then doing:
```
> p = Spree::Product.last
> v = p.variants.build
> v.unit_value = Float::NAN
> v.save
```
...returns true 🤔 
```
> v.reload
> v.unit_value
```
... returns `NaN`
```
> v.save
```
returns `false`
```
> v.errors
@messages={:unit_value=>["is not a number"]}
```
🤯

Reply:
Mystery aside, I added a unit test dec6d2189fd1ec443c9a038947613b4a308c8348 and also a notification to Bugsnag 5dcda317d28285da209d8a80828d794b101e2e3b if we do try to set `unit_value` to `NaN`.

Reply:
I was wondering about that. `NaN` is still a Float and therefore seen as number, apparently. I guess that we would need to do this:

```rb
validates :unit_value, numericality: { other_than: Float::NAN }
```

Anyway, your addition of Bugsnag meets our need to find the source of the problem.

Reply:
Ah yeah that's a good use of that option. The part that was most puzzling was that it was valid at first, but then reloading and saving again made it invalid somehow... but anyway, yes, let's get this out there.